crd/src/lib.rs

```diff
@@ -273,3 +274,133 @@ impl SparkVersion {
         Ok(to_version < from_version)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use stackable_spark_test_utils::cluster::{Load, TestSparkCluster};
+
+    fn setup() -> SparkCluster {
+        TestSparkCluster::load()
+    }
+
+    fn get_instances(node: &SparkNode) -> usize {
+        let mut instances = 0;
+        for selector in &node.selectors {
+            instances += selector.instances;
+        }
+        instances
+    }
+
+    #[test]
+    fn test_spec_hashed_selectors() {
+        let cluster: SparkCluster = setup();
+        let spec: &SparkClusterSpec = &cluster.spec;
+        let cluster_name = &cluster.metadata.name.unwrap();
+
+        let all_spec_hashed_selectors = spec.get_hashed_selectors(cluster_name);
+        if spec.history_server.is_some() {
```

Comment:
You only have a single test-file, no?
How could this ever be different at the moment?

Reply:
Right now its fixed but the plan is to have multiple test files / test clusters.